clients/src/main/java/org/apache/kafka/common/memory/SimpleMemoryPool.java

```diff
@@ -114,6 +119,7 @@ public boolean isOutOfMemory() {
     //allows subclasses to do their own bookkeeping (and validation) _before_ memory is returned to client code.
     protected void bufferToBeReturned(ByteBuffer justAllocated) {
         this.allocateSensor.record(justAllocated.capacity());
```

Comment:
Is the metrics from this allocateSensor not good enough?
If so, can you please explain why?

Reply:
This sensor relies on the `Stat` class, and it appears that it can record statistics like `max`, `sum`, `avg`, and not the entire histogram.